fsevents.js

```diff
@@ -40,25 +53,18 @@ function watch(path) {
   }
 }
 
-function inherit(ctor, superCtor) {
-  ctor.super_ = superCtor;
-  var proto = Object.create(superCtor.prototype, {
-    constructor: {
-      value: ctor,
-      enumerable: false,
-      writable: true,
-      configurable: true
+function proxies(ctor, target) {
+  for (var key in target.prototype) {
+    if (typeof target.prototype[key] !== 'function') {
+      continue;
     }
-  });
-  Object.keys(ctor.prototype).forEach(function(key) {
-    Object.defineProperty(proto, key, {
-      value: ctor.prototype[key],
-      writable: true,
-      enumerable: true,
-      configurable: true
-    });
-  });
-  ctor.prototype = proto;
+
+    ctor.prototype[key] = function() {
+      console.log('invoke', key);
```

Comment:
@pipobscure could you please remove it yourself, or shall I send another pull request?